proof/courier.go

```diff
@@ -91,12 +92,72 @@ type CourierCfg struct {
 	LocalArchive Archiver
 }
 
+// CourierConnStatus is an enum that represents the different states a courier
+// connection can be in.
+type CourierConnStatus int
```

Comment:
Is there a specific reason we need to duplicate this type here? Why can't we just use `grpcconn.State` directly?

Reply:
Ah, I see it's because we don't have a "disconnected" state in `grpcconn.State`. Might be worth mentioning here.